Vagrantfile

```diff
@@ -140,123 +150,76 @@ Vagrant.configure("2") do |config|
-                            machines["m4"]["name"] => {
-                                "build" => {
-                                    "context" => "../../docker/#{machines['m4']['name']}"
-                                },
-                                "command" => ["/usr/sbin/sshd", "-D" ],
-                                "image" => "scz:#{machines['m4']['name']}",
-                                "hostname" => "#{machines['m4']['name']}.#{domain}",
-                                "networks" => {
-                                    "scznet" => {
-                                        "ipv4_address" => machines["m4"]["ip"]
-                                    }
-                                }
-                            },
-                            machines["m5"]["name"] => {
-                                "build" => {
-                                    "context" => "../../docker/#{machines['m5']['name']}"
-                                },
-                                "command" => ["/usr/sbin/sshd", "-D" ],
-                                "image" => "scz:#{machines['m5']['name']}",
-                                "hostname" => "#{machines['m5']['name']}.#{domain}",
-                                "networks" => {
-                                    "scznet" => {
-                                        "ipv4_address" => machines["m5"]["ip"]
-                                    }
-                                }
-                            },
-                            machines["m6"]["name"] => {
-                                "build" => {
-                                    "context" => "../../docker/#{machines['m6']['name']}"
-                                },
-                                "command" => ["/usr/sbin/sshd", "-D" ],
-                                "image" => "scz:#{machines['m6']['name']}",
-                                "hostname" => "#{machines['m6']['name']}.#{domain}",
-                                "networks" => {
-                                    "scznet" => {
-                                        "ipv4_address" => machines["m6"]["ip"]
-                                    }
-                                }
-                            }
-                        },
-                        "networks" => {
-                            "scznet" => {
-                                "driver" => "bridge",
-                                "ipam" => {
-                                    "config" => [{
-                                        "subnet" => "172.20.1.0/24",
-                                        "gateway" => "172.20.1.1"
-                                    }]
-                                }
-                            }
-                        }
-                    }
+
+                config.vm.provision :shell do |shell|
+                    shell.inline = "echo 'Waiting 20 seconds to allow all containers to boot';sleep 20"
```

Comment:
Is it possible to only add the wait for Docker and not for libvirt and virtualbox?

Reply:
I tried and I failed. Determining the provider and performing actions on that takes place in a completely different step than provisioning. If I mark this out for only the Docker provider, I can't use the shell provisioner. If I execute a direct command shell instead, it is executed while building the docker containers and not just before provisioning. I can only add this as a provisioner and in that case I don't know the virtualisation method.
Alternatively, we could add a step to Ansible in the beginning to wait if ansible_virtualization_type=='docker'. But I presume you don't like that :)

This problem can occur in those situations as well. Vagrant boots boxes for the other virtualisation solutions simultaneously and if, for some reason, those solutions cause one of the boxes to linger during boot while the m6 (client) is already up and into the Ansible step, provisioning goes awry. But I concede that this problem occurs mainly because of the build process of Docker and when the containers only boot up, this hardly ever occurs.

I'll remove the wait step. It works correctly most of the time and if it does not work, we don't know how long to wait for it to work properly anyway. It would be nice if Vagrant had some kind of ping-back for its hosts in multiple-host setups, but it doesn't. You need to 'vagrant up --provider=docker --no-provision' and 'vagrant up --provision' in two steps to ensure this.